src/array.rs

```diff
@@ -1510,14 +1624,42 @@ macro_rules! from_slice_impl {
 
 from_slice_impl!(i8, i16, i32, i64, u8, u16, u32, u64, f16, bf16, f32, f64);
 
-impl<T: Into<IValue> + Clone> From<&[T]> for IArray {
-    default fn from(other: &[T]) -> Self {
+// Specific implementation for extending with IValue - no specialization needed
+impl Extend<IValue> for IArray {
+    fn extend<T: IntoIterator<Item = IValue>>(&mut self, iter: T) {
+        let iter = iter.into_iter();
+        self.reserve(iter.size_hint().0);
+        for v in iter {
+            self.push(v);
+        }
+    }
+}
+
+// Specific implementations for IValue (heterogeneous arrays) - no specialization needed
+impl From<Vec<IValue>> for IArray {
+    fn from(other: Vec<IValue>) -> Self {
+        let mut res = IArray::with_capacity(other.len());
+        res.extend(other.into_iter());
+        res
+    }
+}
+
+impl From<&[IValue]> for IArray {
+    fn from(other: &[IValue]) -> Self {
         let mut res = IArray::with_capacity(other.len());
-        res.extend(other.iter().cloned().map(Into::into));
+        res.extend(other.iter().cloned());
         res
     }
 }
```

Comment:
No. We don't want `From<&[Ivalue]>`, we want `From<&[impl Into<IValue>]>`. That's the whole point of having generics in the first place. This PR doesn't fix the problem, it merely deletes functionality.